### PR TITLE
Added support for polynomials with negative degree

### PR DESCRIPTION
**Pull request by CodeMaxx**
- [x] Makes polynomials like `x**(-1) + 2*x**(-2)` valid
- [x] Changed printing for negative exponent terms from `x**-1` to `x**(-1)`
- [x] Add more tests for derivative of negative exponent polynomials etc.

@isuruf @Sumith1896

### CODE REVIEW

**Comment by CodeMaxx** [comment]
@isuruf Any specific tests you think should be added?

**Comment by irislq** [comment]
@isuruf @Sumith1896 Looks like some builds failed because of cmake install issues. Will restarting them solve this?

**Comment by CodeMaxx** [comment]
Yes it will. See this https://travis-ci.org/CodeMaxx/symengine/builds/119995574

**Comment by irislq** [comment]
@isuruf If we can get this merged ASAP, it'll help the UnivariateSeries PR greatly.

**Comment by irislq** [comment]
@CodeMaxx Merge conflicts, can we get this fixed? Thanks!

**Comment by CodeMaxx** [comment]
Fixed

File: symengine/polynomial.cpp [hunk 1]
```diff
@@ -536,14 +536,18 @@ Expression UnivariatePolynomial::max_coef() const
...
+    for (auto it = dict_.rbegin(); it != dict_.rend(); ++it) {
+        x_pow = pow_ex(x, last_deg - it->first);
+        last_deg = it->first;
+        result = it->second + expand(x_pow * result);
     }
-    return ans;
+    x_pow = pow_ex(x, last_deg);
+    result = expand(result * x_pow);
+
+    return result;
```
Comment (isuruf) [inline symengine/polynomial.cpp]: I;m not really sure this would be faster for symbolic polynomials. Also, `expand` calls can be expensive

Comment (CodeMaxx) [inline symengine/polynomial.cpp]: `expand` is absolutely necessary if we want to use Horner's scheme. What do you suggest for this then?

Comment (isuruf) [inline symengine/polynomial.cpp]: Keep it as it was?

Comment (CodeMaxx) [inline symengine/polynomial.cpp]: Thats one options. Or we can benchmark it first and then decide?

Comment (CodeMaxx) [inline symengine/polynomial.cpp]: btw I commited this change by mistake. This is part of the Horner's Scheme PR

Comment (isuruf) [inline symengine/polynomial.cpp]: Let me give you an example where it will surely be slower.
Consider the below example of a polynomial in `x`
`((1+y)**1000)*x + 1`
Evaluating the polynomial at `x=1` will take more time in your method.

Comment (CodeMaxx) [inline symengine/polynomial.cpp]: Right.... so we don't want the coefficients of x to expand. I don't think this can be selectively done with `expand`? Otherwise we can keep the previous implementation.

**Comment by irislq** [comment]
@CodeMaxx Merge conflicts again.

**Comment by CodeMaxx** [comment]
Fixed

**Comment by isuruf** [comment]
+1 to merge after tests pass.
Sorry about the merge conflicts.
